core-ui/src/components/Predefined/Create/DNSEntries/DNSNameRef.js

```diff
@@ -1,13 +1,21 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useGetList } from 'react-shared';
 import { useTranslation } from 'react-i18next';
 
 import { ResourceForm } from 'shared/ResourceForm/ResourceForm';
 
+let memoizedProviders = null;
 export function DNSNameRef({ required, domain, onChange }) {
```

Comment:
can't make a proper suggestion for the whole component, but this should make it handle ResourceForm properties correctly:
```
export function DNSNameRef({ required, ...props }) {
  const { t } = useTranslation();
  const url = `/apis/dns.gardener.cloud/v1alpha1/dnsproviders`;
  const { data } = useGetList()(url);
  useEffect(() => {
    if (data) {
      memoizedProviders = data;
    }
  }, [data]);

  const providers = memoizedProviders || data;

  const domains = (providers || [])
    .map(resource => resource.spec.domains?.include)
    ?.flat();
  const includedDomains = domains.map(domain => ({
    key: domain,
    text: domain,
  }));

  return (
    <ResourceForm.FormField
      {...props}
      required={required}
      label={t('dnsentries.labels.dns-name')}
      propertyPath="$.spec.dnsName"
      input={ResourceForm.ComboboxInput}
      options={includedDomains}
      selectionType="manual"
      key="dns-name-select"
      placeholder={t('dnsentries.placeholders.dns-name')}
    />
  );
}
```
key points:
1. pass incoming props to FormField
2. don't override `ResourceForm.ComboboxInput` - let it handle data from `input={}` by itself
3. `ResourceForm` will take care of `onChange`, `setValue`, etc.
4. any props passed to FormField but not handled get propagated to input (to organize them nicely, i like to place those _after_ the input prop)